src/test/testing_assert.cpp

```diff
@@ -41,47 +43,54 @@ Matrix _table_to_matrix(const opossum::Table& t) {
   return matrix;
 }
 
-void _print_matrix(const Matrix& m) {
-  std::cout << "-------------" << std::endl;
+std::string _matrix_to_string(const std::string& title, const Matrix& m) {
+  std::stringstream stream;
+  stream << "-------" << title << "-------" << std::endl;
   for (unsigned row = 0; row < m.size(); row++) {
     for (opossum::ColumnID col{0}; col < m[row].size(); col++) {
-      std::cout << std::setw(8) << m[row][col] << " ";
+      stream << std::setw(8) << m[row][col] << " ";
     }
-    std::cout << std::endl;
+    stream << std::endl;
   }
-  std::cout << "-------------" << std::endl;
+  stream << "---------------------------" << std::endl;
+  return stream.str();
 }
 }  // namespace
 
 namespace opossum {
 
-::testing::AssertionResult check_table_equal(const Table& tleft, const Table& tright, bool order_sensitive,
-                                             bool strict_types) {
-  Matrix left = _table_to_matrix(tleft);
-  Matrix right = _table_to_matrix(tright);
-
-  const auto print_tables = [&]() {
-    std::cout << "== Tables are not equal ==" << std::endl;
-    _print_matrix(left);
-    _print_matrix(right);
-    std::cout << "==========================" << std::endl;
+bool check_table_equal(const std::shared_ptr<const Table>& opossum_table,
+                       const std::shared_ptr<const Table>& expected_table, OrderSensitivity order_sensitivity,
+                       TypeCmpMode type_cmp_mode, FloatComparisonMode float_comparison_mode) {
+  auto opossum_matrix = _table_to_matrix(opossum_table);
+  auto expected_matrix = _table_to_matrix(expected_table);
+
+  const auto generate_table_comparison = [&]() {
```

Comment:
`diff_tables`?

Reply:
Would expect unix-diff like output then and we don't deliver that (yet)

Reply:
Ok. I don't like the current name too much, but it's not something worth having a discussion over.